src/RFC-0122_Burning.md

```diff
@@ -0,0 +1,252 @@
+    pub total_kernel_offset: BlindingFactor,
+    /// Sum of script offsets for all kernels in this block.
+    pub total_script_offset: BlindingFactor,
+    /// Sum of all burned output commitments in this block.
+    pub burned_total: Commitment,
+    /// Nonce increment used to mine this block.
+    pub nonce: u64,
+    /// Proof of work summary
+    pub pow: ProofOfWork,
+}
+```
+
+This field must contain the sum of all the commitments of all outputs in the block marked as spent. 
+
+$$
+\begin{align}
+burned\_total = \sum_j\mathrm{C_{j}}  \\\\
+\text{for burned output}, i, \\\\
+\end{align}
+\tag{1}
+$$
+
+### Creation of burned output
+
+When creating the output, we need to be able to generate a Proof of Burn for the burned output. 
+In the header, we already commit to each and every [UTXO] ever created. There already is a proof of the existence
+for this burned output, but this proof is only for the complete output and not just the commitment. 
+
+To get around this, we enforce default values for most of the output except the version and commitment field. 
+
+And the output, when burned, MUST have the following field values
+* Empty script,
+* Empty covenant,
+* and empty encrypted_value.
+
+The OutputFeatures MUST be set to the following values:
+* output_type must be set to Burned,
+* maturity must be 0,
+* metadata must be empty
+* and sidechain_features must be empty.
+
+If any of these are not set correctly for a burned output, the output should be dropped as an invalid transaction. 
+
+### Block propagation consensus rules changes
+
+When a block is received, a base node will typically check the following equation to ensure that the emission is correct:
+$$
+\begin{align}
+&\sum_i\mathrm{Cout_{i}} - \sum_j\mathrm{Cin_{j}} + \text{fees} \cdot H - \sum_k\mathrm{K_k} - \text{offset} \stackrel{?}{=}  V_l\cdot H\\\\
+& \text{for each output}, i, \\\\
+& \text{for each input}, j, \\\\
+& \text{for each kernel excess}, k \\\\
+& \textit{offset }\text{is the total kernel offset} \\\\
+& \text{and }V_l \text{ is the block reward for block } l\\\\
+\end{align}
+\tag{2}
+$$
+
+This equation needs to be modified to allow checking the burned number as well, so we change it to:
+$$
+\begin{align}
+&\sum_i\mathrm{Cout_{i}} + \sum_m\mathrm{Cburn_{m}} - \sum_j\mathrm{Cin_{j}} + \text{fees} \cdot H - \sum_k\mathrm{K_k} - \text{offset} \stackrel{?}{=}  V_l\cdot H\\\\
+& \text{for each output}, i, \\\\
+& \text{for each burned output}, m, \\\\
+& \text{for each input}, j, \\\\
+& \text{for each kernel excess}, k \\\\
+& \textit{offset }\text{is the total kernel offset} \\\\
+& \text{and }V_l \text{ is the block reward for block } l\\\\
+\end{align}
+\tag{3}
+$$
+
+We make the following additions to the consensus rules that block propagation must be checked.
+
+When a node receives a block with the node:
+* MUST for each output marked as burned, flag that output as spent immediately upon inserting it as an output. 
```

Comment:
This is implicitly taken care of in the overall balance (5). As it's an implementation detail, it's not a MUST do.

Reply:
This has an effect on the MMR of the outputs. 
If an output is marked as spent, then that reflects on the bitmap of all spent outputs. And this gets committed to in the MMR of the outputs.